File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/dataexport/DataExportDecorator.java
// DataExportDecorator.java

package com.bigboxcode.designpattern.decorator.dataexport;

public abstract class DataExportDecorator implements DataExport {
    protected final DataExport dataExporter;

    public DataExportDecorator(DataExport dataExporter) {
        this.dataExporter = dataExporter;
    }

    @Override
    public void processData() {
        this.dataExporter.processData();
    }
}

File: design-patterns/php/decorator/dataexport/DataExportDecorator.php
<?php
// DataExportDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\DataExport;


class DataExportDecorator implements DataExport {
    public function __construct(protected DataExport $dataExporter) {
    }

    public function processData(): void {
        $this->dataExporter->processData();
    }
}